### PR TITLE
Seed wallet and check balance before a adding a transaction.

### PR DESCRIPTION
**Pull request by nowshad-sust**

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -11,6 +11,9 @@ const myWalletAddress = myKey.getPublic('hex');
 // Create new instance of Blockchain class
 const savjeeCoin = new Blockchain();
 
+// Create a negative transaction which will act as the initial balance
+const tx1 = new Transaction(myWalletAddress, 'address2', -500);
+
 // Create a transaction & sign it with your key
 const tx1 = new Transaction(myWalletAddress, 'address2', 100);
```
Comment (hasezoey) [inline src/main.js]: please remove this transaction or rename the name of the constant

File: src/blockchain.js [hunk 2]
```diff
@@ -186,6 +186,11 @@ class Blockchain {
     if (!transaction.isValid()) {
       throw new Error('Cannot add invalid transaction to chain');
     }
+    
+    // Making sure that the amount sent is not greater than existing balance
+    if (this.getBalanceOfAddress(transaction.fromAddress) < transaction.amount) {
+      throw new Error("Not enough balance");
```
Comment (hasezoey) [inline src/blockchain.js]: please make the quotes to to single quotes, for a consistent style

**Comment by Savjee** [comment]
I agree with the feedback of @hasezoey 
Could you fix these issues and make sure that the Travis build doesn't fail? Then I will happily merge the changes in!

**Comment by hasezoey** [comment]
and still, why the inital negative transaction?

**Comment by nowshad-sust** [comment]
@hasezoey I've made a little change, removed the negative transaction. Added a seedWallet method in Blockchain class.

File: src/blockchain.js [hunk 3]
```diff
@@ -194,6 +194,16 @@ class Blockchain {
 
     this.pendingTransactions.push(transaction);
   }
+  
+  /**
+   * Adds an initial amount to the target address
+   * @param {string} walletNumber
```
Comment (hasezoey) [inline src/blockchain.js]: please add an trailing line, for code style (travis)

**Comment by hasezoey** [comment]
and when this above is resolved, i dont know if it is in scope to have this "joining reward", but i will still approve when it is ready to be merged

File: src/blockchain.js (hunk 3 above)
Comment (Savjee) [reply src/blockchain.js]: I don't think that real blockchain implementations have a seeding function, so we shouldn't do it either. The only way to get coins is to work for them. 

Maybe we can move this function to the test suite? It should set the difficulty very low and mine a few blocks for the given wallet. I think that makes more sense. What's your opinion?

Comment (nowshad-sust) [reply src/blockchain.js]: Yeah, moving it to the test suite seems to be a better choice at this phase of the project.

**Comment by nowshad-sust** [comment]
@hasezoey Facing a problem moving seed function to the test set. Can you give me a few ideas? cc @Savjee

**Comment by hasezoey** [comment]
> @hasezoey Facing a problem moving seed function to the test set. Can you give me a few ideas? cc @Savjee

and what is the problem? can you describe it?

**Comment by nowshad-sust** [comment]
> and what is the problem? can you describe it?
Yeah, not sure where to add the seeding function.

**Comment by Savjee** [comment]
It's okay. Your changes to the main code are already good!
I made a new branch `pr-checking-balance`. Could you change your pull request so it merges with that branch instead of the master?

Then I will correct the tests

**Comment by Savjee** [comment]
Closing this in favor of https://github.com/Savjee/SavjeeCoin/pull/26

**Pull request closed without merging**